File: problems/Codeforces/734A/solution.cpp
#include <iostream>
#include <iterator>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    string _,s;
    cin >> _ >> s;
    int anton = 0, danik = 0;
    for(auto c : s) {
        if(c == 'A') anton++;
        else danik++;
    }

    cout << (anton > danik ? "Anton" : anton < danik ? "Danik" : "Friendship");
    return 0;
}

File: problems/codeforces/734A/solution.rs
use std::io::{self, Read};

fn main() -> io::Result<()> {
    let mut input = String::new();
    io::stdin().lock().read_to_string(&mut input)?;

    let mut anton = 0;
    let mut danik = 0;
    let s = input.split_ascii_whitespace().skip(1).next().unwrap();
    for c in s.chars() {
        if c == 'A' {
            anton += 1; 
        } else {
            danik += 1;
        }
    }

    let result;
    if anton > danik {
        result = "Anton";
    } else if anton < danik {
        result = "Danik";
    } else {
        result = "Friendship";
    }
    print!("{result}");
    Ok(())
}
